### PR TITLE
Topological testing

### PR DESCRIPTION
**Pull request by Jailander**

### CODE REVIEW

**Comment by marc-hanheide** [comment]
is this the right branch? `topological-testing`? When will we merge this with the main branch?

**Comment by Jailander** [comment]
this is the branch on which we are testing the new definitions, I will push some more changes to it today, and @bfalacerda should test before we merge, but testing here seems to be going fine

File: strands_navigation_msgs/msg/Edge.msg [hunk 1]
```diff
@@ -6,9 +6,8 @@ string map_2d
 bool use_default_recovery
 string[] monitor_recovery_pairs_names
 strands_navigation_msgs/DynClassLoaderDef[] monitor_states
-strands_navigation_msgs/DynClassLoaderDef[] recovery_state_machines
 bool use_default_nav_recovery
```
Comment (bfalacerda) [inline strands_navigation_msgs/msg/Edge.msg]: why did you remove this one? its needed to define recovery behaviours for bumper, magnetic strip, etc

Comment (Jailander) [inline strands_navigation_msgs/msg/Edge.msg]: yes It was an accident, I was doing some testing changing the messages for the migration and I left it changed sorry, however this takes me to the next issue, how about we put all this things (except for the booleans) in a JSON/Yaml string, wouldn't it be more comprehensible? @bfalacerda

Comment (bfalacerda) [inline strands_navigation_msgs/msg/Edge.msg]: I don't know, is it something that people usually do?

Comment (Jailander) [inline strands_navigation_msgs/msg/Edge.msg]: well, no but it would make the message a lot simpler, well is just a thought I'll leave it like this for now and look into it afterwards

Comment (bfalacerda) [inline strands_navigation_msgs/msg/Edge.msg]: yeah, if you think it's better I don't mind. If you do change, maybe we can also remove the bools, and assume default to be true iff the yaml string does not have an entry regarding that particular recover

**Comment by marc-hanheide** [comment]
Does this still need merging? I presume not really, right?

**Comment by bfalacerda** [comment]
it does, this is the branch with the new topological map definitions. It's still being tested though, when it's ready I'll merge this PR, and then we have to merge `topological-testing` branch into the default branch. Here we also have hydro-devel as default, but there is an indigo-devel branch already

**Comment by marc-hanheide** [comment]
OK, changed the default to indigo-devel now
